README.md

```diff
@@ -93,20 +93,38 @@ repositories are defined in your build.gradle.
 
 ## Dependencies on other Jenkins Plugins
 
-If your plugin depends on other Jenkins plugins you can specify the dependencies in the following way:
+If your plugin depends on other Jenkins plugins, you can use the same _configurations_ as in Gradle's `java-libary` plugin.
+See [the documentation](https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation) for details on the difference of `api` and `implementation` dependencies.
+For _optional dependencies_, you can use Gradle's [feature variants](https://docs.gradle.org/current/userguide/feature_variants.html).
 
+You can define both dependencies to Jenkins plugins and plain Java libraries.
+The JPI plugin will figure out what you are depending on and process it accordingly (Java libraries will be packaged in the your Jenkins plugin's hpi/jpi file).
+
+The additional `jenkinsServer` configuration can be used to install extra plugins for the `server` task (see below).
+
+Examples:
+
+    java {
+        // define features for 'optional dependencies'
+        registerFeature('ant') {
+            usingSourceSet(sourceSets.main)
+        }
+    }
+    
     dependencies {
-        jenkinsPlugins 'org.jenkinsci.plugins:git:1.1.15'
-        optionalJenkinsPlugins 'org.jenkins-ci.plugins:ant:1.2'
-        jenkinsTest 'org.jenkins-ci.main:maven-plugin:1.480'
```

Comment:
I added a question below about feature variants, but in any case I'd like to see these configurations stick around a while longer. Are there preferred patterns for deprecating and bridging to new configurations?

The reason I'd like to keep them around is to ease the upgrade path. A [search of jenkinsci's gradle wrappers](https://github.com/search?l=&p=1&q=distributionUrl+user%3Ajenkinsci+filename%3Agradle-wrapper.properties&type=Code) shows very few have made it to Gradle 5.x yet, and almost none have made it to 6.x. My experience has been the harder it is to upgrade, the less likely it is to happen. The plugin taking on some additional complexity to support lower effort for consumers seems like a worthwhile exploration to me.

Reply:
With the decision to move to Gradle 6.x as minimum, do you still want to do anything here?

Reply:
I'd like to hold off on doing anything here for now. For next steps, I'm thinking:

1. Merge this PR
2. Publish a release candidate
3. Try to convert some consumers over the next few weeks and use that experience to make a decision about what (if anything) needs to be added to make upgrading easier before publishing a final version.

I'll plan on merging this week unless there are any objections.